src/utils/quote_server_client.ts

```diff
@@ -39,6 +43,18 @@ export class QuoteServerClient {
                 throw new Error('Fee in response is not equal to fee in request');
             }
 
+            if (response.data.signedOrderHash !== payload.orderHash) {
```

Comment:
@phil-ociraptor I also added this validation, which makes sure that the right order hash is returned.